pkg/deploy/lattice/target_group_manager.go

```diff
@@ -160,13 +160,31 @@ func (s *defaultTargetGroupManager) Delete(ctx context.Context, targetGroup *lat
 	}
 	glog.V(6).Infof("TG manager list, listReq %v\n", listTargetsInput)
 	listResp, err := vpcLatticeSess.ListTargetsAsList(ctx, &listTargetsInput)
-	glog.V(6).Infof("manager delete,  listResp %v, err: %v \n", listResp, err)
+	glog.V(6).Infof("TG manager delete,  listResp %v, err: %v \n", listResp, err)
 	if err != nil {
 		return err
 	}
 
 	var targets []*vpclattice.Target
 	for _, t := range listResp {
+		if t.Status != nil && *t.Status != vpclattice.TargetStatusUnused {
```

Comment:
Please add unit test to cover these

Reply:
Will create a issue and add it in a seperate PR